imports/plugins/core/hydra-oauth/server/util/hydra.js

```diff
@@ -78,12 +78,36 @@ function deleteUserSession(id) {
     });
 }
 
+/**
+ * @name refreshAuthToken
+ * @method
+ * @private
+ * @param  {String} options options
+ * @return {Object|String} API res
+ */
+function refreshAuthToken({ refreshToken, clientId, clientSecret }) {
+  return fetch(`${HYDRA_TOKEN_URL}`, {
+    headers: { "Content-Type": "application/x-www-form-urlencoded" },
+    method: "POST",
+    body: `grant_type=refresh_token&refresh_token=${refreshToken}&response_type=token&client_id=${clientId}&client_secret=${clientSecret}`
+  })
+    .then(async (res) => {
+      if (res.status < 200 || res.status > 302) {
+        const json = await res.json();
+        Logger.error("An error occurred while calling refresh API", json.error_description);
+        return Promise.reject(new Error(json.error_description));
```

Comment:
This is an odd mix of normal promise and async/await. It seems like you could rewrite to be completely async/await:

```js
async function refreshAuthToken({ refreshToken, clientId, clientSecret }) {
  const res = await fetch(`${HYDRA_TOKEN_URL}`, {
    headers: { "Content-Type": "application/x-www-form-urlencoded" },
    method: "POST",
    body: `grant_type=refresh_token&refresh_token=${refreshToken}&response_type=token&client_id=${clientId}&client_secret=${clientSecret}`
  });

  const json = await res.json();

  if (res.status < 200 || res.status > 302) {
    Logger.error("An error occurred while calling refresh API", json.error_description);
    throw new Error(json.error_description);
  }

  return json;
}
```